_clients/javascript/helpers.md

```diff
@@ -0,0 +1,168 @@
+| `refreshOnCompletion` | Whether a refresh should be run on all affected indices at the end of the bulk operation. _Default: alse_                                                                                                                                                                                   |
+| `retries`             | Number of times an operation is retried before `onDrop` is called for that document. _Default: Client `maxRetries`_                                                                                                                                                                         |
+| `wait`                | Milliseconds to wait before retrying an operation. _Default: 5000_                                                                                                                                                                                                                          |
+
+### Examples
+
+#### Index
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return {
+      index: { _index: 'example-index' }
+    }
+  }
+})
+```
+
+#### Index with document overwrite
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return [
+      {
+        index: { _index: 'example-index' }
+      },
+      { ...doc, createdAt: new Date().toISOString() }
+    ]
+  }
+})
+```
+
+#### Create
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return {
+      create: { _index: 'example-index', _id: doc.id }
+    }
+  }
+})
+```
+
+#### Create with document overwrite
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return [
+      {
+        create: { _index: 'example-index', _id: doc.id }
+      },
+      { ...doc, createdAt: new Date().toISOString() }
+    ]
+  }
+})
+```
+
+#### Update
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    // The update operation always requires a tuple to be returned, with the
+    // first element being the action and the second being the update options.
+    return [
+      {
+        update: { _index: 'example-index', _id: doc.id }
+      },
+      { doc_as_upsert: true }
+    ]
+  }
+})
+```
+
+#### Update with document overwrite
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return [
+      {
+        update: { _index: 'example-index', _id: doc.id }
+      },
+      {
+        doc: { ...doc, createdAt: new Date().toISOString() },
+        doc_as_upsert: true
+      }
+    ]
+  }
+})
+```
+
+#### Delete
+
+```
+client.helpers.bulk({
+  datasource: arrayOfDocuments,
+  onDocument (doc) {
+    return {
+      delete: { _index: 'example-index', _id: doc.id }
+    }
+  }
+})
+```
```

Comment:
@robdasilva thank you for contributing this doc. Looks good! I would like to reformat and reword a little bit to parallel our other documentation. Do you mind if I take over this PR to do this or would you like me to comment so you can do it yourself?

Reply:
Either way is fine for me. Happy to help where I can. But absolutely happy to hand this over to you as well. It’s probably quicker for you to change as desired :)

Reply:
OK thank you - I will make the changes.